### PR TITLE
win_service: Add option to force stop or restart

### PR DESCRIPTION
**Pull request by dagwieers**
##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

win_service

##### ANSIBLE VERSION

v2.3

##### SUMMARY

This change adds a boolean option to force stop or restart a service.

### CODE REVIEW

**Comment by dagwieers** [comment]
This relies on #20141

File: lib/ansible/modules/windows/win_service.ps1 [hunk 1]
```diff
@@ -27,6 +27,7 @@ Set-Attr $result "changed" $false;
 $name = Get-Attr $params "name" -failifempty $true
 $state = Get-Attr $params "state" $false
 $startMode = Get-Attr $params "start_mode" $false
+$force = Get-AnsibleParam -obj $params -name "force" -type "bool" -default false
```
Comment (ilpianista) [inline lib/ansible/modules/windows/win_service.ps1]: typo here? (missing a `$` before false)

Comment (dagwieers) [reply lib/ansible/modules/windows/win_service.ps1]: Good catch, fixed !

**Comment by mattclay** [comment]
Closing and re-opening to trigger CI.

**Comment by dagwieers** [comment]
This change relates to #20509 (code cleanup of win_service)

**Comment by gundalow** [comment]
@ilpianista Are you happy with the changes in this PR now?

**Comment by ilpianista** [comment]
yes, looks good to me!

**Comment by nitzmahone** [comment]
Question on `force` for this one: since there's no analogue on *nix `service` to this that we should feel hamstrung by, should we pick a more descriptive name? Tough to come up with something that's concise, correct, and applicable to both the restart and stop cases though... `allow_dependent_services` or `manage_dependent_services` are maybe close, but not quite. Open to suggestions or "you're being neurotic, just merge the damned thing". :) I don't have really strong feelings either way.

**Comment by dagwieers** [comment]
@nitzmahone It maps directly with the -Force option from Windows. I even took their explanation for this option, so I wouldn't make it harder and stick to what's already there and possibly known.

File: test/integration/targets/win_service/tasks/main.yml [hunk 2]
```diff
@@ -194,3 +194,27 @@
...
+  win_service: name=EventSystem state=restarted force=yes
+  register: win_service_eventsystem_restart_force
+
+- name: check that stopping the service again succeeds without changes
+  assert:
+    that:
+      - "win_service_eventsystem_restart_force|changed"
+      - "win_service_eventsystem_restart_force.start_mode == 'auto'"
+      - "win_service_eventsystem_restart_force.state == 'running'"
+
```
Comment (mattclay) [inline test/integration/targets/win_service/tasks/main.yml:220]: CI failure due to `yamllint` issue:

```
2017-01-31 02:03:03 Run command: yamllint test/integration/targets/win_service/tasks/main.yml
2017-01-31 02:03:03 test/integration/targets/win_service/tasks/main.yml
2017-01-31 02:03:03   220:1     error    too many blank lines (1 > 0)  (empty-lines)
2017-01-31 02:03:03 
2017-01-31 02:03:03 ERROR: Command "yamllint test/integration/targets/win_service/tasks/main.yml" returned exit status 1.
```

**Comment by mattclay** [comment]
CI failure due to previously mentioned yamllint error.

Full logs here: https://app.shippable.com/runs/588feef669174c0f000c2023/1/console

**Comment by mattclay** [comment]
CI is also failing due to this error:

```
2017-01-31 02:16:44 TASK [win_service : check that restarting the service failed] ******************
2017-01-31 02:16:44 task path: /root/src/github.com/ansible/ansible/test/integration/targets/win_service/tasks/main.yml:203
2017-01-31 02:16:44 fatal: [windows-2008-SP2]: FAILED! => {
2017-01-31 02:16:44     "failed": true, 
2017-01-31 02:16:44     "msg": "The conditional check 'win_service_eventsystem_restart.state == 'running'' failed. The error was: error while evaluating conditional (win_service_eventsystem_restart.state == 'running'): 'dict object' has no attribute 'state'"
2017-01-31 02:16:44 }
2017-01-31 02:16:44 fatal: [windows-2008-R2_SP1]: FAILED! => {
2017-01-31 02:16:44     "failed": true, 
2017-01-31 02:16:44     "msg": "The conditional check 'win_service_eventsystem_restart.state == 'running'' failed. The error was: error while evaluating conditional (win_service_eventsystem_restart.state == 'running'): 'dict object' has no attribute 'state'"
2017-01-31 02:16:44 }
2017-01-31 02:16:44 fatal: [windows-2012-RTM]: FAILED! => {
2017-01-31 02:16:44     "failed": true, 
2017-01-31 02:16:44     "msg": "The conditional check 'win_service_eventsystem_restart.state == 'running'' failed. The error was: error while evaluating conditional (win_service_eventsystem_restart.state == 'running'): 'dict object' has no attribute 'state'"
2017-01-31 02:16:44 }
2017-01-31 02:16:44 fatal: [windows-2012-R2_RTM]: FAILED! => {
2017-01-31 02:16:44     "failed": true, 
2017-01-31 02:16:44     "msg": "The conditional check 'win_service_eventsystem_restart.state == 'running'' failed. The error was: error while evaluating conditional (win_service_eventsystem_restart.state == 'running'): 'dict object' has no attribute 'state'"
2017-01-31 02:16:44 }
```

Full logs here: https://app.shippable.com/runs/588feef669174c0f000c2023/5/console

File: test/integration/targets/win_service/tasks/main.yml (hunk 2 above)
Comment (dagwieers) [reply test/integration/targets/win_service/tasks/main.yml:220]: Do you think this is time worth spending ? Hunting down an additional blank line at the end of a file.
Whoever thinks this is worthwhile I suggest he does that in his own time instead.

Comment (mattclay) [reply test/integration/targets/win_service/tasks/main.yml:220]: You could bring this up in the next Testing Working Group meeting (https://github.com/ansible/community/blob/master/MEETINGS.md). Although I recommend waiting until the one next week, as @gundalow won't be able to attend the meeting tomorrow.

Comment (dagwieers) [reply test/integration/targets/win_service/tasks/main.yml:220]: I will bring it up in Paris with @gundalow ;-)

**Comment by mattclay** [comment]
CI passed, except for the previously mentioned `yamllint` error.

**Comment by dagwieers** [comment]
@jborean93 Do you part over this ? I noticed your new implementation forces by default ? Is that expected ?

**Pull request closed without merging**